scripts/profiler/processed_events.py

```diff
@@ -27,37 +32,127 @@ def __init__(self):
+        self._write_events_types_json(filename_event_types, csv_hash)
+
+    def write_data_to_files(self, filename_events, filename_event_types):
+        self._write_events_csv(filename_events)
+        csv_hash = ProcessedEvents._calculate_md5_hash_of_file(filename_events)
+        self._write_events_types_json(filename_event_types, csv_hash)
+
+    def read_data_from_files(self, filename_events, filename_event_types):
+        csv_hash1 = self._read_events_types_json(filename_event_types)
+        self._read_events_csv(filename_events)
+        csv_hash2 = ProcessedEvents._calculate_md5_hash_of_file(filename_events)
+        if csv_hash1 != csv_hash2:
+            self.logger.warning("Hash values of csv files do not match")
+            self.logger.warning("Events and descriptions may be inconsistent")
+    @staticmethod
+    def _calculate_md5_hash_of_file(filename):
+        return hashlib.md5(open(filename, 'rb').read()).hexdigest()
+
+    def _init_writing_events_csv(self, filename):
+        try:
+            csvfile = open(filename, 'w', newline='')
+            fieldnames = ['type_id', 'timestamp', 'data', 'proc_start_time', 'proc_end_time']
+            wr = csv.DictWriter(csvfile, delimiter=',',
+                                fieldnames=fieldnames)
+            wr.writeheader()
+            return csvfile
+        except IOError:
+            self.logger.error("Problem with accessing file: " + filename)
+            sys.exit()
+
+    def _write_events_csv(self, filename):
+        try:
+            with open(filename, 'w', newline='') as csvfile:
+                fieldnames = ['type_id', 'timestamp', 'data', 'proc_start_time', 'proc_end_time']
+                wr = csv.DictWriter(csvfile, delimiter=',',
+                                    fieldnames=fieldnames)
+                wr.writeheader()
+                for ev in self.tracked_events:
+                    wr.writerow({
+                                 'type_id': ev.submit.type_id,
+                                 'timestamp': ev.submit.timestamp,
+                                 'data': ev.submit.data,
+                                 'proc_start_time': ev.proc_start_time,
+                                 'proc_end_time': ev.proc_end_time
+                                })
+        except IOError:
+            self.logger.error("Problem with accessing file: " + filename)
+            sys.exit()
+
+    def _read_events_csv(self, filename):
+        try:
+            with open(filename, 'r', newline='') as csvfile:
+                rd = csv.DictReader(csvfile, delimiter=',')
+                for row in rd:
+                    type_id = int(row['type_id'])
+                    timestamp = float(row['timestamp'])
+                    # reading event data from single row in csv file
+                    if row['data'][1:-1] != '':
+                        data = list(map(lambda s, data_type: s if data_type == "s" else int(s),
```

Comment:
Use serialization API from Event class to avoid code duplication.

Reply:
Actually the order of data in string and csv file is not the same. In string the data is separated to avoid writing a list because deserialization of list is problematic (to write a list to a csv file it has to be earlier stringified)